src/gui/widgets/PopupHelpWidget.cpp

```diff
@@ -65,10 +60,10 @@ void PopupHelpWidget::setPosition(Qt::Corner corner)
 
 bool PopupHelpWidget::eventFilter(QObject* obj, QEvent* event)
 {
-    if (obj == parentWidget() && event->type() == QEvent::FocusOut) {
-        hide();
-    } else if (obj == m_appWindow && (event->type() == QEvent::Move || event->type() == QEvent::Resize)) {
-        if (isVisible()) {
+    if (isVisible()) {
+        if (obj == parentWidget() && event->type() == QEvent::FocusOut && qApp->focusWindow() != windowHandle()) {
+            hide();
+        } else if (obj == m_appWindow && (event->type() == QEvent::Move || event->type() == QEvent::Resize)) {
```

Comment:
Combine conditions or negate and return early.
To make the lines shorter and the intent clear, this whole thing could even be reformulated in terms of `event->type()` and everything else as secondary conditions.

Reply:
I don't see the problem, we only want to listen to the MainWindow move/resize events in this block. It is distinct from the block above where we are listening for our parent widget's events. This is the most compact and readable the code can get IMO.

Reply:
@phoerious any comment?

Reply:
I'd see the event type as the main condition and visibility as a secondary decision criterion. But it's your call, in the end I'm fine with either.